packages/blade/src/components/List/ListItem.tsx

```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components';
+import React from 'react';
+import { Text } from '../Typography';
+import type { IconComponent } from '../Icons';
+import { useTheme } from '../BladeProvider';
+import { useListContext } from './ListContext';
+import { UnorderedItemIcon } from './ListItemIcons';
+import { ListItemElement } from './ListItemElement';
+import {
+  listItemBulletPaddingRight,
+  listItemBulletMarginTop,
+  listItemOrderedBulletBoxSize,
+  listItemPaddingBottom,
+  listItemPaddingLeft,
+} from './listTokens';
+import type { ListProps } from './List';
+import { getOrderedListItemBullet } from './getOrderedListItemBullet';
+import Box from '~components/Box';
+import { getIn, isValidAllowedChildren, metaAttribute, MetaConstants } from '~utils';
+
+type ListItemProps = {
+  /**
+   * Children to be rendered for ListItem. This can be a text, ListItemLink or another List.
+   *
+   */
+  children: React.ReactNode;
+  /**
+   * Icon to be rendered for a ListItem's bullet.
+   *
+   */
+  icon?: IconComponent;
+  /**
+   * This is a private prop to be used only for internal logic purposes.
+   *
+   */
+  _itemNumber?: undefined;
+};
+
+const StyledListItem = styled(ListItemElement)<{
+  level?: number;
+  variant: NonNullable<ListProps['variant']>;
+  hasIcon: boolean;
+}>(({ level, theme, variant, hasIcon }) => ({
+  paddingLeft: level
+    ? getIn(
+        theme,
+        listItemPaddingLeft[
+          `${variant}${variant === 'unordered' && hasIcon ? 'WithIcon' : ''}` as NonNullable<
+            ListProps['variant'] | 'unorderedWithIcon'
+          >
+        ][level],
+      )
+    : 0,
+}));
+
+const ListItem = ({ children, icon: Icon, _itemNumber }: ListItemProps): React.ReactElement => {
+  const { level, size, icon: ListContextIcon, variant } = useListContext();
+  const { theme, platform } = useTheme();
+  const ItemIcon = Icon ?? ListContextIcon;
+
+  if (level && level > 3) {
+    throw new Error('[Blade List]: List Nesting is allowed only upto 3 levels.');
+  }
+
+  const childrenArray = React.Children.toArray(children);
+
+  const childItem = childrenArray.filter((child) =>
+    isValidAllowedChildren(child, 'List') ? null : child,
```

Comment:
what if i pass something else than List and text? that will also be considered as a valid list item based on this logic but we shall prevent that IMO

Reply:
Yes, prevented that now. It will throw an error